src/sentry/workflow_engine/endpoints/validators/base/detector.py

```diff
@@ -55,8 +58,61 @@ def data_source(self) -> BaseDataSourceValidator:
     def data_conditions(self) -> BaseDataConditionValidator:
         raise NotImplementedError
 
-    def update(self, instance, validated_data):
-        raise NotImplementedError
+    def update_data_conditions(self, instance: Detector, data_conditions: list[DataConditionType]):
+        """
+        Update the data condition if it already exists, create one if it does not
+        """
+        if instance.workflow_condition_group:
+            try:
+                data_condition_group = DataConditionGroup.objects.get(
+                    id=instance.workflow_condition_group.id
+                )
+            except DataConditionGroup.DoesNotExist:
+                raise serializers.ValidationError("DataConditionGroup not found, can't update")
```

Comment:
🤔 could this be agnostic to the parent data model and more generic?

maybe we could make a method that takes the data condition group and then updates accordingly? then we could reuse this for workflow / action / detectors.